packages/react-charting/src/components/HorizontalBarChart/HorizontalBarChart.base.tsx

```diff
@@ -81,108 +80,112 @@ export class HorizontalBarChartBase extends React.Component<IHorizontalBarChartP
-          if (points.chartData && points.chartData![0] && points.chartData![0].horizontalBarChartdata!.x) {
-            datapoint = points.chartData![0].horizontalBarChartdata!.x;
-          } else {
-            datapoint = 0;
-          }
-          points.chartData![1] = {
-            legend: '',
-            horizontalBarChartdata: {
-              x: points.chartData![0].horizontalBarChartdata!.y - datapoint!,
-              y: points.chartData![0].horizontalBarChartdata!.y,
-            },
-            color: palette.neutralLight,
-          };
-
-          // Hide right side text of chart title for absolute-scale variant
-          const chartDataText =
-            this.props.variant === HorizontalBarChartVariant.AbsoluteScale ? null : this._getChartDataText(points!);
-          const bars = this._createBars(points!, palette);
-          const keyVal = this._uniqLineText + '_' + index;
-          const classNames = getClassNames(this.props.styles!, {
-            theme: this.props.theme!,
-            width: this.props.width,
-            showTriangle: !!points!.chartData![0].data,
-            variant: this.props.variant,
-          });
-
-          return (
-            <div key={index}>
-              <div className={classNames.items}>
-                <FocusZone direction={FocusZoneDirection.horizontal}>
-                  <div className={this._classNames.chartTitle}>
-                    {points!.chartTitle && (
-                      <FocusableTooltipText
-                        className={this._classNames.chartTitleLeft}
-                        content={points!.chartTitle}
-                        accessibilityData={points!.chartTitleAccessibilityData}
-                      />
-                    )}
-                    {chartDataText}
-                  </div>
-                </FocusZone>
-                {points!.chartData![0].data && this._createBenchmark(points!)}
-                <FocusZone direction={FocusZoneDirection.horizontal} className={this._classNames.chartWrapper}>
-                  <svg ref={this.barChartSvgRef} className={this._classNames.chart} aria-label={points!.chartTitle}>
-                    <g
-                      id={keyVal}
-                      key={keyVal}
-                      ref={(e: SVGGElement) => {
-                        this._refCallback(e, points!.chartData![0].legend);
-                      }}
-                      // NOTE: points.chartData![0] contains current data value
-                      onClick={() => {
-                        const p = points!.chartData![0];
-                        if (p && p.onClick) {
-                          p.onClick();
-                        }
-                      }}
-                    >
-                      {bars}
-                    </g>
-                  </svg>
-                </FocusZone>
+      <ErrorBoundary handleError={this.props.handleError} theme={this.props.theme}>
```

Comment:
handle isChartEmpty state internally in error boundary control

Reply:
done